satellite/audit/service.go

```diff
@@ -72,6 +78,31 @@ func (service *Service) Run(ctx context.Context) (err error) {
 	})
 }
 
+// RemoteSegment takes a remote segment found in metainfo and creates a reservoir for it if it doesn't exist already
+func (service *Service) RemoteSegment(ctx context.Context, path storj.Path, pointer *pb.Pointer) (err error) {
+	defer mon.Task()(&ctx, path)(&err)
+
+	remote := pointer.GetRemote()
+	pieces := remote.GetRemotePieces()
+
+	for i, piece := range pieces {
+		if _, ok := service.Reservoirs[piece.NodeId]; !ok {
+			reputable, err := service.overlay.IsVetted(ctx, piece.NodeId)
```

Comment:
Since we are already adding logic here to determine whether the node is reputable, I think it makes sense to also pick the vetted vs. unvetted reservoir count here, and pass it into `NewReservoir`. Mentioned in my other comment.